crypto/bn/bn_lib.c

```diff
@@ -294,15 +289,17 @@ BIGNUM *BN_copy(BIGNUM *a, const BIGNUM *b)
     if (b->top > 0)
         memcpy(a->d, b->d, sizeof(b->d[0]) * b->top);
 
-    a->top = b->top;
     a->neg = b->neg;
+    a->top = b->top;
+    a->flags |= b->flags & BN_FLG_FIXED_TOP;
```

Comment:
This should have a comment.

Reply:
This should go to #6662, so I'm adding commit there.